cosmos/providers/dbt/core/profiles/base.py

```diff
@@ -0,0 +1,167 @@
+"""
+This module contains a base class that other profile mappings should
+inherit from to ensure consistency.
+"""
+from __future__ import annotations
+
+from logging import getLogger
+from typing import Any
+
+import yaml
+from typing_extensions import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from airflow.models import Connection
+
+logger = getLogger(__name__)
+
+
+class BaseProfileMapping:
+    """
+    A base class that other profile mappings should inherit from to ensure consistency.
+    Responsible for mapping Airflow connections to dbt profiles.
+    """
+
+    airflow_connection_type: str = "generic"
+    is_community: bool = False
+
+    required_fields: list[str] = []
+    secret_fields: list[str] = []
+    airflow_param_mapping: dict[str, str | list[str]] = {}
+
+    def __init__(self, conn: Connection, profile_args: dict[str, Any] | None = None):
+        self.conn = conn
+        self.profile_args = profile_args or {}
+
+    def can_claim_connection(self) -> bool:
+        """
+        Return whether the connection is valid for this profile mapping.
+        """
+        if self.conn.conn_type != self.airflow_connection_type:
+            return False
+
+        for field in self.required_fields:
+            try:
+                if not getattr(self, field):
+                    logger.info(
+                        "Not using mapping %s because %s is not set",
+                        self.__class__.__name__,
+                        field,
+                    )
+                    return False
+            except AttributeError:
+                logger.info(
+                    "Not using mapping %s because %s is not set",
+                    self.__class__.__name__,
+                    field,
+                )
+                return False
+
+        return True
+
+    def get_profile(self) -> dict[str, Any]:
+        """
+        Return a dbt profile based on the Airflow connection.
+        """
+        raise NotImplementedError
+
+    def get_env_vars(self) -> dict[str, str]:
```

Comment:
WDYT if we had used Python `@property` for getters which don't need a parameter, such as:
* `get_profile` -> `profile`
* `get_env_vars` -> `env_vars`